docs/styleguide-cpp.md

```diff
@@ -1,320 +1,660 @@
+| Number | Item                              | Writing Convention                  | Example                                 |
+| 1      | Class names                       | PascalCase                          | MyCoolExample                           |
+| 2      | Struct names                      | PascalCase                          | MyCoolExample                           |
+| 3      | Variable names                    | Lower snake_case                    | my_cool_example                         |
+| 4      | Function Variable names           | Lower snake_case (`p_` prefix)      | p_my_cool_example                       |
+| 5      | Private Variable names            | Lower snake_case (`_` suffix)       | \_my_cool_example                       |
+| 6      | Function and Method names         | Lower snake_case                    | my_cool_example                         |
+| 7      | Private Function and Method names | Lower snake_case (`_` suffix)       | \_my_cool_example                       |
+| 8      | Constants                         | Upper SNAKE_CASE                    | MY_COOL_EXAMPLE                         |
+| 9      | Macros                            | Upper SNAKE_CASE                    | MY_COOL_EXAMPLE                         |
+| 10     | Class or container type aliases   | PascalCase                          | MyCoolExample                           |
+| 11     | Builtin or numeric type aliases   | PamelCase ( `_t` suffix)            | MyCoolExample_t                         |
+| 12     | Enum names                        | PascalCase                          | MyCoolExample                           |
+| 13     | Enum values                       | Upper SNAKE_CASE                    | MY_COOL_EXAMPLE                         |
+| 14     | Filenames for classes             | PascalCase following the class name | MyCoolExample.hpp/MyCoolExample.cpp     |
+| 15     | Filenames                         | Lower snake_case                    | my_cool_example.hpp/my_cool_example.cpp |
+
+### 1.4. Do’s and Don’ts
+
+#### 1.4.1. Keep the opening scope brace on the same line
+
+```c++
+//Correct
+if (condition) {
+    //Cool things here
+}
+
+//Incorrect
+if (condition)
+{
+    //cool things here
+}
+```
+
+#### 1.4.2. Keep spaces between arithmetic operators, relational operators, logical operators, variables, and literals
+
+```c++
+//Correct
+int my_number = (some_variable + 10) % 3;
+
+//Incorrect
+int my_number=(some_variable+10)%3;
+```
+
+#### 1.4.3. Prefix and postfix operators should be immediately attached to the variable
+
+```c++
+//Correct
+for (int x = 0; x < 10; x++) {
+    //Something cool
+}
+
+//Incorrect
+for (int x = 0; x < 10; x ++ ) {
+    //Something cool
+}
+```
+
+#### 1.4.4. Avoid `inline`, use Godot’s `_FORCE_INLINE_` for release inline behavior and `_ALWAYS_INLINE_` for always inline behavior
```

Comment:
What about non godot c++ code?

Reply:
Game engine reliance is inevitable for use of a game engine.

Reply:
Semantically some things in C++ need to be marked inline even if you really don't care if the compiler actually compiles it as inline, and so you explicitly don't want the inlining to be forced. It is also nicer to do constexpr inline ... with global constants to prevent them from being materialized by taking an address

Reply:
Aside from Godot-specific types/macros polluting the C++ codebase more than necessary, the ability to conditionally inline functions is not necessary.

The rationale given for conditional inlining is that it makes code easier to debug, however inlined methods should only be used for trivial (or near-trivial) attribute accessors.

Reminder that most of the team's developers have intermediate-to-expert levels of C++ experience.

Reply:
None of this addresses this issue and by now referring to "near-trivial" usage you've given even better justification for why conditional inline matters, if use of inline matters, there is no excuse not to rely on conditional inline as even in trivial cases it will make debug more difficult.